packages/web3-eth-contract/src/contract.ts

```diff
@@ -215,6 +218,182 @@ const contractSubscriptions = {
 	newBlockHeaders: NewHeadsSubscription,
 };
 
+
+/*
+ * This class is only supposed to be used for the return of `new Contract(...).deploy(...)` method.
+ */
+export class DeployerMethodClass<FullContractAbi extends ContractAbi> {
```

Comment:
why there is need of moving `deploy` to different class?

Reply:
(from usability perspective for external users)

Reply:
Good question,
I needed to add `populateTransaction`. But the `Contract` class was already about 1500 lines of code. So abstracting the logic into multiple classes will ease code maintenance for us. And if a user wanted to extend the `Contract` class he will have to deal with simpler code structures... 
And we would in the future also create a class for the return of the method `_createContractMethod` as it also returns a class like structure.